File: 4 - Class vs. Instance/Solution.java
import java.util.Scanner;

public class Solution {
    static class Person {
        private int age;

        public Person(int initialAge) {
            // Add some more code to run some checks on initialAge
            if (initialAge > 0) {
                age = initialAge;
            } else {
                System.out.println("Age is not valid, setting age to 0.");
                age = 0;
            }
        }

        public void amIOld() {
            // Write code determining if this person's age is old and print the correct statement:
            if (age < 13) System.out.println("You are young.");
            else if (age < 18) System.out.println("You are a teenager.");
            else System.out.println("You are old.");
        }

        public void yearPasses() {
            // Increment this person's age.
            age++;
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        for (int i = 0; i < T; i++) {
            int age = sc.nextInt();
            Person p = new Person(age);
            p.amIOld();
            for (int j = 0; j < 3; j++) {
                p.yearPasses();
            }
            p.amIOld();
            System.out.println();
        }
        sc.close();
    }
}

File: 4 - Class vs. Instance/Solution.ts
class Person {
  age: number;
  constructor(initialAge: number) {
    if (initialAge > 0) {
      this.age = initialAge;
    } else {
      console.log("Age is not valid, setting age to 0.");
      this.age = 0;
    }
  }
  public amIOld() {
    // Write code determining if this person's age is old and print the correct statement:
    if (this.age < 13) console.log("You are young.");
    else if (this.age < 18) console.log("You are a teenager.");
    else console.log("You are old.");
  }

  public yearPasses() {
    // Increment the age of the person in here
    this.age++;
  }
}

class Solution {
  public static Main() {
    let T: number = 25;
    for (let i = 0; i < T; i++) {
      let age = 20;
      let p = new Person(age);
      p.amIOld();
      for (let j = 0; j < 3; j++) {
        p.yearPasses();
      }
      p.amIOld();
    }
  }
}

Solution.Main();
